### PR TITLE
test: Storage Provider/Store mock

### PR DESCRIPTION
**Pull request by rolsonquadras**
- New Storage provider and Store mock
- Moved outbound provider mock to /internal/mock folder

Cloes #202 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: pkg/didcomm/protocol/didexchange/service_test.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/dispatcher"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/decorator"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
-	mocktransport "github.com/hyperledger/aries-framework-go/pkg/internal/didcomm/transport/mock"
```
Comment (troyronda) [inline pkg/didcomm/protocol/didexchange/service_test.go]: do we really want all mocks in a single folder?

Comment (fqutishat) [reply pkg/didcomm/protocol/didexchange/service_test.go]: it's easier to find in one folder

Comment (rolsonquadras) [reply pkg/didcomm/protocol/didexchange/service_test.go]: Well, we can argue on both sides though. Personally, i feel folder structure becomes too complicated if we have a folder for each mock. anyways, the mocks are inside internal folder.

Comment (troyronda) [reply pkg/didcomm/protocol/didexchange/service_test.go]: even if you don’t want one folder per package, are we sure we shouldn’t separate mocks at the top level … e.g., didcomm

Comment (fqutishat) [reply pkg/didcomm/protocol/didexchange/service_test.go]: I think we should go with  pkg/internal/mocks/didcomm

Comment (rolsonquadras) [reply pkg/didcomm/protocol/didexchange/service_test.go]: @troyronda @fqutishat may be going down one level makes sense, i guess.

reg name - i feel putting mock at higher level than feature will give a good readability. People can go to mocks folder and see variable feature level mocks available.

Comment (rolsonquadras) [reply pkg/didcomm/protocol/didexchange/service_test.go]: [Updated](https://github.com/hyperledger/aries-framework-go/compare/257b15506fd69db7afc2e44467e044d9f3af5a18..9e430efccae39bf51273eafe671bffd5120dec70)

**Comment by troyronda** [comment]
Should be rebased on master @rolsonquadras

**Pull request merged**